vm-setup/roles/v1aX_integration_test/templates/metal3datatemplate-template.yaml

```diff
@@ -0,0 +1,69 @@
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+kind: Metal3DataTemplate
+metadata:
+  name: ${ CLUSTER_NAME }-{{ template_type }}-template
+spec:
+  clusterName: ${ CLUSTER_NAME }
+  metaData:
+    objectNames:
+      - key: name
+        object: machine
+      - key: local-hostname
+        object: machine
+      - key: local_hostname
+        object: machine
+    ipAddressesFromIPPool:
+      - key: provisioningIP
+        name: provisioning-pool
+    prefixesFromIPPool:
+      - key: provisioningCIDR
+        name: provisioning-pool
+  networkData:
+    links:
+      ethernets:
+{% for nic_name in bmh_nic_names %}
+        - type: phy
+          id: {{ nic_name }}
+          macAddress:
+            fromHostInterface: {{ nic_name }}
+{% endfor %}
+{% if EXTERNAL_VLAN_ID != "" %}
+      vlans:
+        - id: "vlan{{ EXTERNAL_VLAN_ID }}"
+          mtu: 1500
+          macAddress:
+            fromHostInterface: {{ bmh_nic_names | first }}
+          vlanID: {{ EXTERNAL_VLAN_ID }}
+          vlanLink: {{ bmh_nic_names | first }}
```

Comment:
We have to pick a link (interface) on which to create a vlan interface, it will be something like `eno1` or `eth0` or `enp0s1`, depending on the system and what 'Predictable Network Naming' comes out with. The vlan interface gets the name `eth0.3` or similar. When the bare metal IP / keepalived IP is added to this vlan interface, traffic will be tagged with the vlan id.